packages/coding-standard/src/Fixer/Commenting/RemoveCommentedCodeFixer.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Fixer\Commenting;
+
+use Nette\Utils\Strings;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+use Symplify\CodingStandard\Fixer\AbstractSymplifyFixer;
+use Symplify\RuleDocGenerator\Contract\DocumentedRuleInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Symplify\CodingStandard\Tests\Fixer\Commenting\RemoveCommentedCodeFixer\RemoveCommentedCodeFixerTest
+ */
+final class RemoveCommentedCodeFixer extends AbstractSymplifyFixer implements DocumentedRuleInterface
+{
+    /**
+     * @see https://regex101.com/r/uMDMtt/3
+     * @var string
+     */
+    private const COMMENTED_CODE_REGEX = '#\/\/\s+(\$.*|[^"].*\(.*?\));$#';
```

Comment:
This should cover any commented code.

E.g:

```php
-// private const SOME_CONST = '...';
```

```php
-// class SomeClass
-// {
-// }
```

Etc.

I guess better would be to parse token with `token_get_all()` and if it's valid, then it should be removed.

They use this approach in the sniff: https://github.com/squizlabs/PHP_CodeSniffer/blob/99b0339bce7eaae8b5285398d5f1492da734d1ac/src/Standards/Squiz/Sniffs/PHP/CommentedOutCodeSniff.php#L76-L173

Reply:
In the end, this test file should be covered - https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/Squiz/Tests/PHP/CommentedOutCodeUnitTest.inc

Reply:
I will try

Reply:
I tried to replicate with `token_get_all()` function, loop it : 

```php
        $tokens = token_get_all('<?php '. file_get_contents($file->getRealPath()) .' ?>');

        foreach ($tokens as $token) {
        // ...
```

but it seems `$token['code']`, `$token['line']`, `$token['content']` never catched. The `token_get_all()` get array like the following:

```php
 array(3) {
    [0]=>
    int(380)
    [1]=>
    string(33) "// | A comment block           |
"
    [2]=>
    int(61)
  }
```

@TomasVotruba any idea?

Reply:
@TomasVotruba I think we can discuss about this tomorrow.